macos_defaults.py

```diff
@@ -79,14 +79,16 @@ class Actions:
 
     def selected_text() -> str:
         try:
-            return ui.focused_element().AXSelectedText
-        except Exception:
-            try:
-                # ui.focused_element() sometimes returns NoElement.
-                # https://github.com/talonvoice/talon/issues/480
+            selected_text = ui.focused_element().AXSelectedText
+            if not selected_text:
+                # Some partially-accessible applications incorrectly report empty selections sometimes, and this can be
+                # quite bad depending on the use case. For maximum safety we have to fall back to the clipboard
+                # implementation in this case. This could be customized if we knew the application was fully
+                # trustworthy.
                 #
-                # TODO(pcohen): extract this focused_element() -> AXFocusedUIElement fallback
-                # if we expect to need it in the future.
-                return ui.active_app().element.AXFocusedUIElement.AXSelectedText
-            except Exception:
+                # See https://github.com/phillco/talon-axkit/issues/59
                 return actions.next()
+
+            return selected_text
+        except Exception:
```

Comment:
except: should be fine here.

Reply:
I think using a raw `except` is an anti pattern, right? https://wiki.python.org/moin/HandlingExceptions

Reply:
Yes, but it's no more an anti-pattern than `except Exception`, and in fact I'd argue that `except Exception` kind of lives in a no-man's land.  I would argue that you should either except specific exceptions, or if you want a general catch-all, which it seems you want here, you should use `except:`.  I'm not sure why you'd want to pass through exceptions that do not derive from `Exception`

Reply:
> But I'm basing most of this on that link that you sent

You're right; it's not the best reference for this argument.

Reply:
I’m convinced. Keep it as it is.

Reply:
Oh interesting thanks TIL. I'm convinced too